src/frontend/js/hooks/useStepManager/index.spec.ts

```diff
@@ -0,0 +1,129 @@
+    };
+    const { result } = renderHook(() => useStepManager(manifest));
+
+    // - step 0
+    expect(result.current.step).toEqual('step0');
+
+    // - step 1
+    act(() => result.current.next());
+    expect(result.current.step).toEqual('step1');
+
+    // - Reset: should go back to step0
+    act(() => result.current.reset());
+    expect(result.current.step).toEqual('step0');
+  });
+
+  it('triggers onEnter and onExit hooks on step transition', () => {
+    const manifest: Manifest = {
+      start: 'step0',
+      steps: {
+        step0: {
+          next: 'step1',
+          onEnter: jest.fn(),
+          onExit: jest.fn(),
+        },
+        step1: {
+          next: null,
+          onEnter: jest.fn(),
+          onExit: jest.fn(),
+        },
+      },
+    };
+    const { result } = renderHook(() => useStepManager(manifest));
+
+    // - step 0
+    expect(result.current.step).toEqual('step0');
+    expect(manifest.steps.step0.onEnter).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step0.onExit).not.toHaveBeenCalled();
+    expect(manifest.steps.step1.onEnter).not.toHaveBeenCalled();
+    expect(manifest.steps.step1.onExit).not.toHaveBeenCalled();
+
+    // - step 1
+    act(() => result.current.next());
+
+    expect(result.current.step).toEqual('step1');
+    expect(manifest.steps.step0.onEnter).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step0.onExit).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step1.onEnter).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step1.onExit).not.toHaveBeenCalled();
+
+    // - terminated state
+    act(() => result.current.next());
+
+    expect(result.current.step).toEqual(null);
+    expect(manifest.steps.step0.onEnter).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step0.onExit).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step1.onEnter).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step1.onExit).toHaveBeenCalledTimes(1);
+
+    // - Reset: go back to step0
+    act(() => result.current.reset());
+
+    expect(manifest.steps.step0.onEnter).toHaveBeenCalledTimes(2);
+    expect(manifest.steps.step0.onExit).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step1.onEnter).toHaveBeenCalledTimes(1);
+    expect(manifest.steps.step1.onExit).toHaveBeenCalledTimes(1);
+
+    // - step 1
+    act(() => result.current.next());
+
+    expect(manifest.steps.step0.onEnter).toHaveBeenCalledTimes(2);
+    expect(manifest.steps.step0.onExit).toHaveBeenCalledTimes(2);
+    expect(manifest.steps.step1.onEnter).toHaveBeenCalledTimes(2);
+    expect(manifest.steps.step1.onExit).toHaveBeenCalledTimes(1);
+
+    // - Reset: go back to step0
+    act(() => result.current.reset());
+
+    expect(manifest.steps.step0.onEnter).toHaveBeenCalledTimes(3);
+    expect(manifest.steps.step0.onExit).toHaveBeenCalledTimes(2);
+    expect(manifest.steps.step1.onEnter).toHaveBeenCalledTimes(2);
+    expect(manifest.steps.step1.onExit).toHaveBeenCalledTimes(2);
+  });
+});
```

Comment:
```suggestion
});
//two suggested tests so that we know component still works and we are aware of this two behaviour
//i'm wondering if Manifest shouldn't be more controlled for the last test to not be possible
it('can contain a single step', () => {
    const manifest: Manifest = {
      start: 'step0',
      steps: {
        step0: {
          next: null,
        }
      },
    };
    const { result } = renderHook(() => useStepManager(manifest));

    // - step 0
    expect(result.current.step).toEqual('step0');

    // - terminated state
    act(() => result.current.next());
    expect(result.current.step).toEqual(null);

    // - Reset: should go back to step0
    act(() => result.current.reset());
    expect(result.current.step).toEqual('step0');
    
  });
  
  it("loads manifest if first step doesn't exist but steps aren't linked anymore", () => {
    const manifest: Manifest = {
      start: 'step0',
      steps: {
        step1: {
          next: 'step2',
          onEnter: jest.fn(),
          onExit: jest.fn(),
        },
        step2: {
          next: null,
          onEnter: jest.fn(),
          onExit: jest.fn(),
        },
      },
    };
    const { result } = renderHook(() => useStepManager(manifest));

    // - step 0
    expect(result.current.step).toEqual('step0');
    expect(manifest.steps.step0).toEqual(null);
    
    act(() => result.current.next());
    expect(result.current.step).toEqual(null);

    act(() => result.current.reset());
    expect(result.current.step).toEqual('step0');
  });
```

Reply:
Ok for the first test you suggest. About the second, in fact we should not be able to do that. If I can type `Manifest` as discussed in your comment below, we should no longer be able to write it.

Reply:
That would be great and it was the point of writing it!